lib/passport-configurator.js

```diff
@@ -300,15 +300,29 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
                 }
               } else {
                 user.hasPassword(password, function(err, ok) {
-                  if (ok) {
-                    done(null, userProfile);
+                  if (ok) {          
+         if (!user.emailVerified) 
+         {
+              // Fail to log in if email verification is not done yet
+              return done(null, false, {message: 'User Email has not been verified.'});
+              err = new Error('login failed as the email has not been verified');
+              err.statusCode = 401;
+              err.code = 'LOGIN_FAILED_EMAIL_NOT_VERIFIED';
+              fn(err);
+          } 
+          else 
+          {
+                   user.createAccessToken(accessToken);
```

Comment:
Why do we want to create accessToken here? For this path `options.setAccessToken` is not set.